From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Thu, 27 Mar 2008 17:57:07 +0100
Subject: modular braille + speakup support

- Vanilla 2.6.24 has keyboard/VT notifier support
- both braille module and speakup need additional 4 exports to work


diff -puN drivers/char/consolemap.c~basic-braille-screen-reader-support drivers/char/consolemap.c
--- a/drivers/char/consolemap.c~basic-braille-screen-reader-support
+++ a/drivers/char/consolemap.c
@@ -277,6 +277,7 @@ u16 inverse_translate(struct vc_data *co
 			return p->inverse_translations[m][glyph];
 	}
 }
+EXPORT_SYMBOL_GPL(inverse_translate);
 
 static void update_user_maps(void)
 {
diff -puN drivers/char/keyboard.c~basic-braille-screen-reader-support drivers/char/keyboard.c
--- a/drivers/char/keyboard.c~basic-braille-screen-reader-support
+++ a/drivers/char/keyboard.c
@@ -109,6 +109,7 @@ const int max_vals[] = {
 const int NR_TYPES = ARRAY_SIZE(max_vals);
 
 struct kbd_struct kbd_table[MAX_NR_CONSOLES];
+EXPORT_SYMBOL_GPL(kbd_table);
 static struct kbd_struct *kbd = kbd_table;
 
 struct vt_spawn_console vt_spawn_con = {
@@ -259,6 +260,7 @@ void kd_mksound(unsigned int hz, unsigne
 	} else
 		kd_nosound(0);
 }
+EXPORT_SYMBOL_GPL(kd_mksound);
 
 /*
  * Setting the keyboard rate.
diff -puN drivers/char/vt.c~basic-braille-screen-reader-support drivers/char/vt.c
--- a/drivers/char/vt.c~basic-braille-screen-reader-support
+++ a/drivers/char/vt.c
@@ -4003,6 +4003,7 @@ u16 screen_glyph(struct vc_data *vc, int
 		c |= 0x100;
 	return c;
 }
+EXPORT_SYMBOL_GPL(screen_glyph);
 
 /* used by vcs - note the word offset */
 unsigned short *screen_pos(struct vc_data *vc, int w_offset, int viewed)
